InteractiveHtmlBom/ecad/kicad.py

```diff
@@ -526,4 +527,7 @@ def Run(self):
             return
 
         parser = PcbnewParser(pcb_file_name, config, logger, board)
-        ibom.run_with_dialog(parser, config, logger)
+        try:
+            ibom.run_with_dialog(parser, config, logger)
+        except ParsingException as e:
+            logger.error(str(e))
```

Comment:
Use e.message instead of type conversion here and everywhere.

Reply:
This doesn't work:
```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "InteractiveHtmlBom/generate_interactive_bom.py", line 65, in <module>
    exit_error(logger, ExitCodes.ERROR_PARSE, e.message)
AttributeError: 'ParsingException' object has no attribute 'message'
```

Reply:
The `message` attribute is commonly defined by some buit-in exceptions (not all). It just contains the first argument passed to the exception.
The `__str__` attribute returns the same you want from `message`.
And I tried all the `logging` error levels, none prints anything other than the `str`. Not even the `repr`. The following is a test I tried for Python 2.7.16 and 3.7.3:

```
import logging

logging.basicConfig(level=logging.DEBUG)

class ETest(Exception):
    pass


try:
    raise ETest("Hi!")
except ETest as e:
    print(e)
    print(repr(e))
    print(type(e))
    print(e.__dict__)
    logging.debug(e)
    logging.error(e)
    logging.warning(e)
    logging.info(e)
    logging.critical(e)
```
For Python 2 I get:

```
Hi!
ETest('Hi!',)
<class '__main__.ETest'>
{}
DEBUG:root:Hi!
ERROR:root:Hi!
WARNING:root:Hi!
INFO:root:Hi!
CRITICAL:root:Hi!
```

And for Python 3:

```
Hi!
ETest('Hi!')
<class '__main__.ETest'>
{}
DEBUG:root:Hi!
ERROR:root:Hi!
WARNING:root:Hi!
INFO:root:Hi!
CRITICAL:root:Hi!
```

Reply:
Sorry, let me clarify. I only mentioned that loggers print stack trace to show the difference, even so looks like I was wrong. Either way, I don't think we should print the stack trace.

> The message attribute is commonly defined by some buit-in exceptions (not all). It just contains the first argument passed to the exception.

Ugh, just checked, it was removed in python3. py2 used to have message attribute in every Exception derived class.

I think the PR looks fine now, I'll test it and if nothing comes up merge it over the weekend.